### PR TITLE
Search sessions or speakers by voice-search

### LINKED ISSUE #438: Search sessions or speakers by your voice in search screen
## Kind (Required)

- Improvement

 ## Overview (Required)

- Title says everything!

 ## Links	
- Figma
    - https://www.figma.com/file/4r9becvhDy3GfXaXex8E8d/App?node-id=842%3A907

## Image
スクリーンショット 2020-01-19 19 40 42

### PR DESCRIPTION
**Pull request by imsaku**
## Issue
- close #438 

## Overview (Required)
-Apply voice input to SearchView
-Don't apply design about mic-icon #437 

## Links
-https://developer.android.com/reference/android/widget/SearchView
-https://developer.android.com/guide/topics/search/search-dialog.html
-https://developer.android.com/guide/topics/search/searchable-config.html

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
Thanks!
I merged this PR
Can you check after merge master? 🙇 
https://github.com/DroidKaigi/conference-app-2020/pull/781

**Comment by imsaku** [comment]
@takahirom 
Sure. I'll do it!
btw, which should I use "merge" or "rebase" just in case?
Sorry to a basic question.🙇‍♂️

**Comment by takahirom** [comment]
I think both rebase and merge have good points, so I'm not particular about the policy. Please do it any way you like. 😄

**Comment by takahirom** [comment]
Sorry please wait this https://github.com/DroidKaigi/conference-app-2020/pull/787

**Comment by takahirom** [comment]
I merged it 🙇

**Comment by imsaku** [comment]
> I think both rebase and merge have good points, so I'm not particular about the policy. Please do it any way you like. 😄

That's right. Thanks for answer.😳

**Comment by imsaku** [comment]
I think no problem between #781 and mine.
However, I have no confidence about my code💦

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt [hunk 1]
```diff
@@ -148,10 +153,27 @@ class SearchSessionsFragment : Fragment(R.layout.fragment_search_sessions), Inje
         }
     }
 
+    override fun onResume() {
+        super.onResume()
+        val intent = activity!!.intent
```
Comment (takahirom) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: You can use `requireActivity()` 👍

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt [hunk 2]
```diff
@@ -148,10 +153,27 @@ class SearchSessionsFragment : Fragment(R.layout.fragment_search_sessions), Inje
...
+        if (intent.action == Intent.ACTION_SEARCH) {
+            val query: String = intent.getStringExtra(SearchManager.QUERY)!!
```
Comment (takahirom) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]:
```suggestion
            val query: String = requrireNotNull(intent.getStringExtra(SearchManager.QUERY))
```

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt [hunk 3]
```diff
@@ -148,10 +153,27 @@ class SearchSessionsFragment : Fragment(R.layout.fragment_search_sessions), Inje
...
+            val searchResult = searchSessionsViewModel.uiModel.requireValue().searchResult
```
Comment (takahirom) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: I think we don't need to check to change query  😄

Comment (imsaku) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: I thought so.
However,  `onResume` is called after `setQuery` then it's query is same.
I want to only use once `setQuery` but  I have no idea w/o `query != searchResult.query`. 😂

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt (hunk 1 above)
Comment (imsaku) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: Fixed it!😊

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt (hunk 2 above)
Comment (imsaku) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: I took it!
But i didn't notice typo💦
I fixed it 4998c258a09acc3275a89ea01fb4f4f2d5ae33a9

Comment (takahirom) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: 🙇

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt (hunk 3 above)
Comment (takahirom) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: It's not so many times, so I think it's okay to call twice, but is there any problem? 👀

Comment (imsaku) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: Not twice. It loops  `onResume` -> `setQuery` -> `onResume`. 
For example, below code.
```
if (searchView != null) {
                searchView.setQuery(query, true)
            }
```

**Comment by takahirom** [comment]
When we use light theme. we can not see voice icon 😭

**Comment by imsaku** [comment]
Yes. i know it 😂 
I tried to apply voice icon design.
But I stopped it and deleted codes when I noticed #437 ...💦

**Comment by imsaku** [comment]
I can try again about voice icon if you can wait🙇‍♂️

**Comment by takahirom** [comment]
> I can try again about voice icon if you can wait🙇‍♂

Yes! You can this in 1.2 or later 👍

**Comment by takahirom** [comment]
I pushed a little refactoring!

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt [hunk 4]
```diff
@@ -172,17 +173,19 @@ class SearchSessionsFragment : Fragment(R.layout.fragment_search_sessions), Inje
         inflater.inflate(R.menu.menu_search_sessions, menu)
         this.menu = menu
         val searchView = menu.findItem(R.id.search_view).actionView as SearchView
-        val searchManager = activity!!.getSystemService(SEARCH_SERVICE) as SearchManager
-        searchView.setSearchableInfo(searchManager.getSearchableInfo(activity!!.componentName))
+        val context = requireContext()
+        context.getSystemService<SearchManager>()?.let { searchManager ->
+            searchView.setSearchableInfo(searchManager.getSearchableInfo(activity!!.componentName))
```
Comment (takahirom) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: Sorry can you use requireActivity() 🙇

Comment (takahirom) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]:
```suggestion
            searchView.setSearchableInfo(searchManager.getSearchableInfo(requireActivity().componentName))
```

Comment (imsaku) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SearchSessionsFragment.kt]: ahh..sorry, I should have done it🙇‍♂️

**Comment by imsaku** [comment]
> I pushed a little refactoring!

I have many things to learn from this refactoring 👀
Thanks!😊

**Comment by imsaku** [comment]
I pushed about voice icon changes!
But icon color is different from the design 😢 [figma](https://www.figma.com/file/4r9becvhDy3GfXaXex8E8d/App?node-id=842%3A907)

Light | Dark
:--: | :--:
 |

**Comment by takahirom** [comment]
Can you check this?
./gradlew :feature:session:ktlint

**Comment by imsaku** [comment]
> Can you check this?
> ./gradlew :feature:session:ktlint

I fixed 🙏

**Comment by takahirom** [comment]
Thanks! LGTM 👍

**Comment by imsaku** [comment]
Thanks for so kind reviews 😭
It helps me a lot 🙇‍♂️

**Pull request merged**